libs/sdk-common/src/input_parser.rs

```diff
@@ -206,19 +238,38 @@ pub async fn parse(
     let input = input.trim();
 
     // Try to parse the destination as a bip353 address.
-    let input_parsed = match bip353_parse(input, &DNS_RESOLVER).await {
-        Some(value) => value,
-        None => input.to_string(),
+    let (input, is_bip353) = match bip353_parse(input, &DNS_RESOLVER).await {
+        Some(value) => (value, true),
+        None => (input.to_string(), false),
     };
 
-    let input = input_parsed.as_str();
+    if let Ok(input_type) = parse_core(&input).await {
+        let input_type = if is_bip353 {
+            match input_type {
+                #[cfg(feature = "liquid")]
+                InputType::Bolt12Offer { offer } => InputType::Bip353Address {
+                    resolved_bip353: ResolvedBip353::Bolt12Offer {
+                        bip353_address: input,
+                        offer,
+                    },
+                },
+                InputType::LnUrlPay { data } => InputType::Bip353Address {
+                    resolved_bip353: ResolvedBip353::LnUrlPay {
+                        bip353_address: input,
+                        request_data: data,
+                    },
+                },
```

Comment:
This could technically be a Bitcoin address too no?

Reply:
A BIP353 address can technically be one, but we don't support that for now. See https://github.com/breez/breez-sdk-greenlight/issues/1161.